datasets/us/navy/crawler.py

```diff
@@ -0,0 +1,148 @@
+    if match:
+        # Extract the full title (including everything before "Admiral" or "Adm.")
+        title = match.group(1).strip() + " " + match.group(2).strip()
+        # Extract the name (everything after "Admiral" or "Adm.")
+        name = match.group(3).strip()
+    else:
+        # Log a warning if no title is found
+        context.log.warn(f"Failed to extract title from name: {name}")
+        title = None
+    emit_person(context, "us", url, role, name, title=title)
+
+
+def process_page(context: Context, page_number: int):
+    url = API_URL % page_number
+    try:
+        data = context.fetch_json(url)
+    except orjson.JSONDecodeError as e:
+        context.log.error(f"Failed to decode JSON from {url}: {e}")
+        return False, False
+    except Exception as e:
+        context.log.error(f"Failed to fetch JSON from {url}: {e}")
+        return False, False
+
+    if not data:
+        context.log.error(f"No data found for page {page_number}")
+        return False, False
+
+    if isinstance(data, dict):
+        html_data = data.get("data")
+        done = data.get("done", True)
+    else:
+        try:
+            root = etree.fromstring(data)
+            html_data = root.find(".//data").text
+            done = root.find(".//done").text.lower() == "true"
+        except etree.XMLSyntaxError as e:
+            context.log.error(f"Failed to parse XML for page {page_number}: {e}")
+            return False, False
+
+    if not html_data:
+        context.log.error(f"No HTML data found for page {page_number}")
+        return False, done
+
+    items = html.fromstring(html_data).findall(".//li")
+    for item in items:
+        item_html = html.tostring(item, encoding="unicode")
+        crawl_person(context, item_html)
+
+    return True, done
+
+
+def unblock_validator(doc: html.HtmlElement) -> bool:
+    return len(doc.xpath(".//div[contains(@class, 'DNNModuleContent')]")) > 0
+
+
+def parse_html(context):
+    doc = fetch_html(
+        context, context.data_url, unblock_validator=unblock_validator, cache_days=3
+    )
+    for div in doc.xpath(
+        './/div[contains(@class, "DNNModuleContent") and contains(@class, "ModPhotoDashboardC")]'
+    ):
+        leader_divs = div.xpath('.//div[contains(@class, "leader-title")]/a')
+        for leader_div in leader_divs:
+            leader_url = urljoin(BASE_URL, leader_div.get("href"))
+            name_element = leader_div.find(".//h3")
+            role_element = leader_div.find(".//h2")
+            if name_element is None or role_element is None:
+                context.log.info(
+                    f"Skipping incomplete leader entry: {html.tostring(leader_div, pretty_print=True, encoding='unicode')}"
+                )
+                continue
+            name = name_element.text_content().strip()
+            role = role_element.text_content().strip()
+            if not name or not role:
+                continue  # Skip if either name or role is empty
```

Comment:
It doesn't look like this is something that happens so I'm nervous this hides a bug in the future

```suggestion
            assert name and role, f"Name or role missing: {name}, {role}"
```